src/ast.rs

```diff
@@ -53,7 +55,7 @@ impl Program {
         self.blocks.first().expect("logic error")
     }
 
-    pub fn next_block<'a, 'b>(&'a self, block: &'b Block) -> &'a Block {
+    pub fn next_block<'b>(&'a self, block: &'b Block) -> &'a Block {
```

Comment:
That's interesting. Why is `'a` gone in `next_block`?

Reply:
`'a` is now a parameter of the whole `impl`, since the `Program` struct now stores a `Span` and this one is borrows strings, so itself has a param.